### PR TITLE
Refactor Sense pallet.

### PR DESCRIPTION
**Pull request by FiberMan**
A bit of refactoring of Sense pallet:
- one storage for all entity properties
- one extrinsic to update properties
- types extracted to separate file
- tests/benchmarks updated
- renaming, enhancements, etc.

**To be discussed:**
Currently only increments available for properties (XP, Rep, Trust).
I think it would be useful to have decrement functionality.

### CODE REVIEW

**Review by 2075 — APPROVED** [review]
lgtm

**Comment by 2075** [comment]
Thank you for the optimisations and contributing! Almost forgot to submit the comment :)
- I am not sure if u8 is big enough for e.g. xp and rep, eventually a u16 would be more sufficient
- Agree with decrements requirement

File: sense/src/benchmarking.rs [hunk 1]
```diff
@@ -11,21 +11,11 @@ benchmarks! {
...
-		let caller_origin = <T as frame_system::Config>::Origin::from(RawOrigin::Root);
-		Sense::<T>::create_entity(caller_origin, account("1", 0, 0), BoundedVec::truncate_from(vec![1; 1]))?;
-	}: _(RawOrigin::Root, account("1", 0, 0), 255)
-
-	mod_trust {
-		let caller_origin = <T as frame_system::Config>::Origin::from(RawOrigin::Root);
-		Sense::<T>::create_entity(caller_origin, account("1", 0, 0), BoundedVec::truncate_from(vec![1; 1]))?;
-	}: _(RawOrigin::Root, account("1", 0, 0), 255)
-
+	}: _(RawOrigin::Root, account("1", 0, 0), property_type, 255)
```
Comment (vovacha) [inline sense/src/benchmarking.rs:18]: The caller should be whitelisted for DB reads/writes. Same for `create_entity` benchmark.
```
// One way
frame_benchmarking::benchmarking::add_to_whitelist;
// Another way
frame_benchmarking::whitelisted_caller;
```

**Review by vovacha — CHANGES_REQUESTED** [review]
Thank you, good job.
Build and tests are ok. The only thing is a minor fix for the benchmarking required.

File: sense/src/types.rs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+impl<BlockNumber> EntityProperty<BlockNumber> {
+	pub fn new(value: u64, block_number: BlockNumber) -> EntityProperty<BlockNumber> {
+		EntityProperty { value, mutated: block_number }
+	}
+}
+#[derive(Encode, Decode, Clone, PartialEq, Eq, PartialOrd, Ord, TypeInfo, MaxEncodedLen)]
+#[cfg_attr(feature = "std", derive(Debug))]
+pub enum PropertyType {
+	Experience, 
+    Trust,
```
Comment (vasylenko-yevhen) [inline sense/src/types.rs:43]: Spaces vs tabs here.

File: sense/src/types.rs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+#[derive(Encode, Decode, Default, Eq, Copy, PartialEq, Clone, RuntimeDebug, TypeInfo, MaxEncodedLen)]
+pub struct EntityProperty<BlockNumber> {
+	pub(crate) value: u64,
+	pub(crate) mutated: BlockNumber,
+}
+
+impl<BlockNumber> EntityProperty<BlockNumber> {
+	pub fn new(value: u64, block_number: BlockNumber) -> EntityProperty<BlockNumber> {
```
Comment (vasylenko-yevhen) [inline sense/src/types.rs:35]: Seems like `new` is not used and not needed here.

File: sense/src/types.rs [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+	where
+		BlockNumber: Clone,
+	{
+		Entity { account, index, cid, created: block_number.clone(), mutated: block_number }
+	}
+}
+
+#[derive(Encode, Decode, Default, Eq, Copy, PartialEq, Clone, RuntimeDebug, TypeInfo, MaxEncodedLen)]
+pub struct EntityProperty<BlockNumber> {
+	pub(crate) value: u64,
```
Comment (vasylenko-yevhen) [inline sense/src/types.rs:30]: I would also discuss privacy of `Entity` and `EntityProperty` fields. Is there a reason why they have different visibility scope - `private` vs `pub(crate)`?

**Review by vasylenko-yevhen — COMMENTED** [review]
Thanks for your contribution. Looks nice, just few small remarks/questions.

File: sense/src/benchmarking.rs (hunk 1 above)
Comment (vovacha) [reply sense/src/benchmarking.rs:18]: Looking into examples in the Substrate repo, it seems `RawOrigin::Root` doesn't whitelisted from the DB operations. It's not clear why, but I'm going to raise this in substrate StackExchange. For now, we can leave it as is.

File: sense/src/types.rs (hunk 2 above)
Comment (FiberMan) [reply sense/src/types.rs]: Fixed.

File: sense/src/types.rs (hunk 3 above)
Comment (FiberMan) [reply sense/src/types.rs:35]: It is used in tests, but now I fixed also in `lib` to use `new` method.

File: sense/src/types.rs (hunk 4 above)
Comment (FiberMan) [reply sense/src/types.rs]: Indeed, it is not necessary to have public fields there. I created a getter instead, thanks.

File: sense/src/benchmarking.rs (hunk 1 above)
Comment (FiberMan) [reply sense/src/benchmarking.rs:18]: I've checked it and came to the same conclusion :)
thanks!

**Review by vovacha — APPROVED** [review]
LGTM

**Pull request merged**